File: src/test/java/testdata/JsonIgnoreTypeTestPOJO.java
package testdata;

import java.util.List;

public class JsonIgnoreTypeTestPOJO {


    private String username;
    private List<Role> roles;

    @com.fasterxml.jackson.annotation.JsonIgnoreType
    public class Role {
        private String roleName;
        private List<JsonIgnoreTypeTestPOJO> users;
    }
}

File: src/test/java/testdata/kotlin/JsonIgnoreTypeTestPOJO.kt
package testdata.kotlin

import com.fasterxml.jackson.annotation.JsonIgnoreType

class JsonIgnoreTypeTestPOJO {
    private val username: String? = null
    private val roles: List<Role>? = null

    @JsonIgnoreType
    inner class Role {
        private val roleName: String? = null
        private val users: List<JsonIgnoreTypeTestPOJO>? = null
    }
}